spec/factories/documents.rb

```diff
@@ -25,38 +25,35 @@
 
 FactoryBot.define do
   factory :document do
-    document do
-      Rack::Test::UploadedFile.new(
-        File.expand_path('features/examples/longer_lorem.pdf', Rails.root),
-        'application/pdf'
-      )
+    transient do
+      file_type { :pdf }
+      content_type { { pdf: 'application/pdf', docx: 'application/msword' }[file_type] }
+      sequence(:filename) { |n| "testfile#{n}.#{file_type}" }
+      file_source do
+        {
+          pdf: 'features/examples/longer_lorem.pdf',
+          docx: 'features/examples/shorter_lorem.docx'
+        }[file_type]
+      end
```

Comment:
unless this is necessary fo this fix can you put this in a separate PR, as it seems like scope creep. The use of simple traits seems simpler in any event.

Reply:
I did this mainly so that I could set the filename, to ensure that two files had the same name. I'll see if I can restore the traits.